client/internal/proxy/direct.go

```diff
@@ -0,0 +1,57 @@
+package proxy
+
+import (
+	log "github.com/sirupsen/logrus"
+	"net"
+)
+
+// DirectNoProxy is used when there is no need for a proxy between ICE and WireGuard.
+// This is possible in either of these cases:
+// - peers are in the same local network
+// - one of the peers has a public static IP (host)
+// DirectNoProxy will just update remote peer with a remote host and fixed WireGuard port (r.g. 51820).
+// In order DirectNoProxy to work, WireGuard port has to be fixed for the time being.
+type DirectNoProxy struct {
+	config Config
```

Comment:
I see on several place in this repo to use a huge "config" struct instead of pass arguments via constructor. This is a really bad pattern because for the developer hard to understand the scope of the current checked code and logic of the system. In the current situation the DirectNoProxy does not use all parameters of Config. So why we store it in memory? If we check deeper the proxy.Config we can realize  that we completely eliminate it because almost the all information what carry is initialized on during the process's startup time and it will never changing but we duplicate these information for every peer.

So I think this is a great time to do a quick and simple refactor.
What I recommend:
- Delete the proxy.Config and pass all necessary arguments to the constructors.
- The PreSharedKey is come from cfg file and it has never changing in runtime. So in this part of the code make no sens to take care about it. Update the iface.wGConfigurer.updatePeer function and remove this argument.
- WgListenAddr probably the same situation. We pass the local wg port within peer.ConnConfig and proxyConfig to the peer.Conn and both config store the same information use only one of them or create a new function in the iface.WGIface for get back this info.

Reply:
We initially used a config struct to avoid constantly changing the signature of methods. In the case of a proxy, this might be unnecessary because there are just a few parameters. So it makes sense to do the refactor you suggested. But I'd rather do it in a separate PR.

Reply:
I understand your point and it is true and good. In case the proxyConfig I think all information is available peer.ConnConfig also.
In a separated pull request the workflow is nicer but it cause extra work for you. I think the feature/bind is already a huge change so does not matter more one extra change. I suppose if you merge the bind into the main branch you never want to revert it any more.